poly-commit/src/utils.rs

```diff
@@ -70,21 +78,37 @@ impl<F: Field> Matrix<F> {
         );
 
         // TODO more efficient to run linearly?
-        let entries: Vec<Vec<F>> = (0..n)
+        let rows: Vec<Vec<F>> = (0..n)
             .map(|row| (0..m).map(|col| entry_list[m * row + col]).collect())
             .collect();
 
-        Self { n, m, entries }
+        Self { n, m, rows }
+    }
+
+    /// Returns self as a list of rows
+    pub(crate) fn rows(&self) -> &Vec<Vec<F>> {
+        &self.rows
+    }
+
+    /// Returns the entry in position (i, j). **Indexing starts at 0 in both coordinates**,
+    /// i.e. the first element is in position (0, 0) and the last one in (n - 1, j - 1),
+    /// where n and m are the number of rows and columns, respectively.
+    ///
+    /// Index bound checks are waived for efficiency and behaviour under invalid indexing is undefined
+    #[cfg(test)]
+    pub(crate) fn entry(&self, i: usize, j: usize) -> F {
+        self.rows[i][j]
     }
 
-    /// Returns a Matrix given a list of its rows, each in turn represented as a list of field elements.
+    /// Returns a RowMajorMatrix given a list of its rows, each in turn represented as a list of field elements.
     ///
     /// # Panics
     /// Panics if the sub-lists do not all have the same length.
-    pub(crate) fn new_from_rows(row_list: Vec<Vec<F>>) -> Self {
-        let m = row_list[0].len();
+    #[cfg(test)]
+    pub(crate) fn new_from_rows(row_major: Vec<Vec<F>>) -> Self {
```

Comment:
This is only used in testing now

Reply:
can be removed / refactor tests

Reply:
Note that `RowMajorMatrix` is used in Hyrax and Ligero too. We need this method for them. So I would say let's keep it.